doc/mm.rst

```diff
@@ -0,0 +1,101 @@
+=======================
+Nim's Memory Management
+=======================
+
+.. default-role:: code
+.. include:: rstcommon.rst
+
+:Author: Andreas Rumpf
+:Version: |nimversion|
+
+..
+
+
+  "The road to hell is paved with good intentions."
+
+
+Introduction
+============
+
+A memory-management algorithm optimal for every use-case cannot exist.
+Nim provides multiple paradigms for needs ranging from large multi-threaded
```

Comment:
just below, it says the other MM's are dead, and there's only one true MM?

Reply:
Yeah, we need to update the text.